holoviews/operation/element.py

```diff
@@ -758,8 +758,12 @@ def _process(self, p, element, ranges={}):
         for d1, d2 in permuted_dims:
             if d1 == d2:
                 if p.diagonal_type is not None:
-                    values = element.dimension_values(d1)
-                    el = p.diagonal_type(values, vdims=[d1])
+                    if p.diagonal_type._1d:
```

Comment:
That does not seem extensible. If I want to declare a new Element type, which has one of these properties how would I tell ``is_distribution1D`` about that? As far as I'm aware this is the only class attribute like this and is defined like this on ``Dataset``:

```
    # In the 1D case the interfaces should not automatically add x-values
    # to supplied data
    _1d = False
```